engine/src/main/java/org/terasology/logic/characters/KinematicCharacterMover.java

```diff
@@ -673,7 +673,13 @@ private void walk(final CharacterMovementComponent movementComp, final Character
             }
         } else {
             if (moveResult.isTopHit() && endVelocity.y > 0) {
-                endVelocity.y = -0.5f * endVelocity.y;
+                if (input.isFirstRun()) {
+                    Vector3f hitVelocity = new Vector3f(state.getVelocity());
+                    hitVelocity.y += (distanceMoved.y / moveDelta.y) * (endVelocity.y - state.getVelocity().y);
+                    logger.debug("Hit at " + hitVelocity);
+                    entity.send(new VerticalCollisionEvent(state.getPosition(), hitVelocity));
+                }
+                endVelocity.y = -0.0f * endVelocity.y;
```

Comment:
@skaldarnar I am keeping the multiplication because if at a later stage a variable Coefficient of Restitution(cor) is introduced than we can just put the variable name instead of zero and the collisions would work like a charm.